sys/include/auto_init_utils.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2022 Otto-von-Guericke-Universität Magdebug
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_auto_init_utils Utilities to influence the order of auto-initialized modules
+ * @ingroup     sys
+ *
+ * @{
+ * @file
+ * @brief       Auto-initialization utilities
+ *
+ * A module priority value in RIOTs auto-initialization sequence is, as of now,
+ * a hard coded 4-digit number, but is likely to be changed and automatically
+ * generated in the future. Modules are initialized in ascending
+ * (alpha)numerical order of their priority, which means that modules with
+ * lower priorities are initialized first, as long as their priorities comply
+ * with the current rules.
+ *
+ * @experimental
+ * @author      Fabian Hüßler <fabian.huessler@ovgu.de>
+ */
+#ifndef AUTO_INIT_UTILS_H
+#define AUTO_INIT_UTILS_H
+
+#include <stdint.h>
+#include "xfa.h"
+#include "macros/xtstr.h"
+#include "kernel_defines.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef CONFIG_AUTO_INIT_ENABLE_DEBUG
+/**
+ * @brief   Print a debug message before a module is initialized
+ */
+#define CONFIG_AUTO_INIT_ENABLE_DEBUG 0
+#endif
+
+/**
+ * @brief   Auto-init function type
+ */
+typedef void (*auto_init_fn_t)(void);
+
+/**
+ * @brief   Data type to store the priority of a module
+ */
+typedef uint16_t auto_init_prio_t;
+
+/**
+ * @brief   Type to represent a module to be auto-initialized
+ */
+typedef struct {
+    auto_init_fn_t init;    /**< Function to initialize the module */
+#if IS_ACTIVE(CONFIG_AUTO_INIT_ENABLE_DEBUG)
+    auto_init_prio_t prio;  /**< Module priority */
+    const char *name;       /**< Module name */
+    const char *extra;      /**< User supplied string */
+#endif
+} auto_init_module_t;
+
+#if IS_ACTIVE(CONFIG_AUTO_INIT_ENABLE_DEBUG) || defined(DOXYGEN)
+/**
+ * @brief   Add a module to the auto-initialization array
+ *
+ * @param   module      Module name
+ * @param   priority    Priority level
+ * @param   function    Function to be called on initialization
+ * @param   ...         Optional debug string
+ */
+#define AUTO_INIT_MODULE(module, priority, function, ...)                               \
+    XFA_CONST(auto_init_xfa, priority)                                                  \
+    auto_init_module_t auto_init_xfa_ ## module                                         \
+        = (auto_init_module_t) { (auto_init_fn_t)function,                              \
+                                 priority, XTSTR(module), ""__VA_ARGS__ }
```

Comment:
```suggestion
        = { .init = (auto_init_fn_t)function,                                           \
            .prio = priority,                                                           \
            .name = XTSTR(module),                                                      \
            .extra = ""__VA_ARGS__ }
```

I don't see why the case is needed, and it actually makes the compilation fail on native.